File: js_ts/Leetcode/169_Majority_Element.ts
function majorityElement(nums: number[]): number {
    let candidate, count = 0, n = nums.length;
    for (let i = 0; i < n; i++) {
        if (count === 0) candidate = nums[i];
        if (nums[i] === candidate) count++;
        else count--;
    }
    return candidate;
};

File: python/Leetcode/169_Majority_Element.py
from typing import List
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        el = nums[0]
        count = 1
        for i in range(1, len(nums)):
            if nums[i] == el:
                count += 1
            elif count == 0:
                el = nums[i]
                count = 1
            else:
                count -= 1
        return el
